tests/test_assemblies.py

```diff
@@ -268,3 +303,156 @@ def test_assembly(self, identifier, num_neuroids, properties, stimulus_set_ident
+    ])
+    def test_stimulus_set_alignment(self, identifier, field):
+        full_name = f"brendel.Geirhos2021_{identifier}"
+        assembly = brainscore.get_assembly(full_name)
+        assert assembly.stimulus_set is not None
+        assert assembly.stimulus_set.identifier == full_name
+        assert set(assembly.stimulus_set[field]) == set(assembly[field].values)
+
+    # test the number of subjects:
+    @pytest.mark.parametrize('identifier, num_subjects', [
+        ('colour', 4),
+        ('contrast', 4),
+        ('cue-conflict', 10),
+        ('edge', 10),
+        ('eidolonI', 4),
+        ('eidolonII', 4),
+        ('eidolonIII', 4),
+        ('false-colour', 4),
+        ('high-pass', 4),
+        ('low-pass', 4),
+        ('phase-scrambling', 4),
+        ('power-equalisation', 4),
+        ('rotation', 4),
+        ('silhouette', 10),
+        ('stylized', 5),
+        ('sketch', 7),
+        ('uniform-noise', 4),
+    ])
+    def test_num_subjects(self, identifier, num_subjects):
+        assembly = brainscore.get_assembly(f"brendel.Geirhos2021_{identifier}")
+        assert len(np.unique(assembly['subject'].values)) == num_subjects
+
+    # test the number of images
+    @pytest.mark.parametrize('identifier, num_images', [
+        ('colour', 1280),
+        ('contrast', 1280),
+        ('cue-conflict', 1280),
+        ('edge', 160),
+        ('eidolonI', 1280),
+        ('eidolonII', 1280),
+        ('eidolonIII', 1280),
+        ('false-colour', 1120),
+        ('high-pass', 1280),
+        ('low-pass', 1280),
+        ('phase-scrambling', 1120),
+        ('power-equalisation', 1120),
+        ('rotation', 1280),
+        ('silhouette', 160),
+        ('stylized', 800),
+        ('sketch', 800),
+        ('uniform-noise', 1280),
+    ])
+    def test_num_images(self, identifier, num_images):
+        assembly = brainscore.get_assembly(f"brendel.Geirhos2021_{identifier}")
+        assert len(np.unique(assembly['image_id'].values)) == num_images
+
+    # tests assembly coords for the 14 "normal" sets:
+    @pytest.mark.parametrize('identifier, length', [
+        ('colour', 5120),
+        ('contrast', 5120),
+        ('eidolonI', 5120),
+        ('eidolonII', 5120),
+        ('eidolonIII', 5120),
+        ('false-colour', 4480),
+        ('high-pass', 5120),
+        ('low-pass', 5120),
+        ('phase-scrambling', 4480),
+        ('power-equalisation', 4480),
+        ('rotation', 5120),
+        ('stylized', 4000),
+        ('sketch', 5600),
+        ('uniform-noise', 5120),
+    ])
+    @pytest.mark.parametrize('field', [
+        'image_id',
+        'image_id_long',
+        'choice',
+        'truth',
+        'condition',
+        'response_time',
+        'trial',
+        'subject',
+        'session',
+    ])
+    def test_fields_present(self, identifier, field, length):
+        assembly = brainscore.get_assembly(f"brendel.Geirhos2021_{identifier}")
+        assert len(assembly[field].values) == length
```

Comment:
this is redundant with the two statements below, I would delete lines 407 through 417 and this one, unless there's something specific to check. If there is only one dimension and the dimension has a certain length, then by definition all coords will be that length too.

Reply:
I think it's a good idea to test that the fields are present, but this would then be a separate test, i.e. separate into two:

```
    @pytest.mark.parametrize('identifier, length', [
        ('colour', 5120),
        ('contrast', 5120),
        ('eidolonI', 5120),
        ('eidolonII', 5120),
        ('eidolonIII', 5120),
        ('false-colour', 4480),
        ('high-pass', 5120),
        ('low-pass', 5120),
        ('phase-scrambling', 4480),
        ('power-equalisation', 4480),
        ('rotation', 5120),
        ('stylized', 4000),
        ('sketch', 5600),
        ('uniform-noise', 5120),
    ])
    def test_length(identifier, length):
        assembly = brainscore.get_assembly(f"brendel.Geirhos2021_{identifier}")
        assert len(assembly['presentation']) == length

    @pytest.mark.parametrize('field', [
        'image_id',
        'image_id_long',
        'choice',
        'truth',
        'condition',
        'response_time',
        'trial',
        'subject',
        'session',
    ])
    def test_fields_present(self, identifier, field, length):
        assembly = brainscore.get_assembly(f"brendel.Geirhos2021_{identifier}")
        assert hasattr(assembly, field)
```